### PR TITLE
[FLINK-3760] Fix StateDescriptor.readObject

### PR DESCRIPTION
**Pull request by aljoscha**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/api/common/state/StateDescriptor.java [hunk 1]
```diff
@@ -284,8 +284,14 @@ private void readObject(final ObjectInputStream in) throws IOException, ClassNot
...
 		if (hasDefaultValue) {
 			int size = in.readInt();
+
 			byte[] buffer = new byte[size];
-			int bytesRead = in.read(buffer);
+
+			int bytesRead = 0;
+
+			while (bytesRead < size) {
+				bytesRead += in.read(buffer, bytesRead, size - bytesRead);
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/api/common/state/StateDescriptor.java]: is there a reason we're not simply using readFully(buffer) ?

Comment (aljoscha) [inline flink-core/src/main/java/org/apache/flink/api/common/state/StateDescriptor.java]: That is true, changing.

**Comment by zentol** [comment]
+1

**Comment by aljoscha** [comment]
Thanks for having a look! I'll merge it...

**Comment by aljoscha** [comment]
Manually merged

**Pull request closed without merging**